dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+     * @param hosts must not be {@literal null} nor empty!
+     * @return new instance of {@link DotReactiveElasticsearchClient}.
+     */
+    public static ReactiveElasticsearchClient create(HttpHeaders headers, String... hosts) {
+
+        Assert.notNull(headers, "HttpHeaders must not be null");
+        Assert.notEmpty(hosts, "Elasticsearch Cluster needs to consist of at least one host");
+
+        ClientConfiguration clientConfiguration = ClientConfiguration.builder().connectedTo(hosts)
+                .withDefaultHeaders(headers).build();
+        return create(clientConfiguration);
+    }
+
+    /**
+     * Create a new {@link DotReactiveElasticsearchClient} given {@link ClientConfiguration}. <br />
+     * <strong>NOTE</strong> If the cluster requires authentication be sure to provide the according {@link HttpHeaders}
+     * correctly.
+     *
+     * @param clientConfiguration Client configuration. Must not be {@literal null}.
+     * @return new instance of {@link DotReactiveElasticsearchClient}.
+     */
+    public static ReactiveElasticsearchClient create(ClientConfiguration clientConfiguration) {
+
+        Assert.notNull(clientConfiguration, "ClientConfiguration must not be null");
+
+        WebClientProvider provider = getWebClientProvider(clientConfiguration);
+
+        HostProvider hostProvider = HostProvider.provider(provider,
+                clientConfiguration.getEndpoints().toArray(new InetSocketAddress[0]));
+        return new DotReactiveElasticsearchClient(hostProvider);
+    }
+
+    private static WebClientProvider getWebClientProvider(ClientConfiguration clientConfiguration) {
+
+        Duration connectTimeout = clientConfiguration.getConnectTimeout();
+        Duration soTimeout = clientConfiguration.getSocketTimeout();
+
+        TcpClient tcpClient = TcpClient.create();
+
+        if (!connectTimeout.isNegative()) {
+            tcpClient = tcpClient.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, Math.toIntExact(connectTimeout.toMillis()));
+        }
+
+        if (!soTimeout.isNegative()) {
+
+            tcpClient = tcpClient.doOnConnected(connection -> connection //
+                    .addHandlerLast(new ReadTimeoutHandler(soTimeout.toMillis(), TimeUnit.MILLISECONDS))
+                    .addHandlerLast(new WriteTimeoutHandler(soTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+        }
+
+        String scheme = "http";
+        HttpClient httpClient = HttpClient.from(tcpClient);
+
+        if (clientConfiguration.useSsl()) {
+
+            httpClient = httpClient.secure(sslConfig -> {
+
+                Optional<SSLContext> sslContext = clientConfiguration.getSslContext();
+                sslContext.ifPresent(it -> sslConfig.sslContext(new JdkSslContext(it, true, ClientAuth.NONE)));
+            });
+
+            scheme = "https";
+        }
+
+        ReactorClientHttpConnector connector = new ReactorClientHttpConnector(httpClient);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'connector' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)